src/inc/corinfo.h

```diff
@@ -2129,6 +2129,12 @@ class ICorStaticInfo
             CORINFO_CLASS_HANDLE        classHnd
             ) { return false; }
 
+    // Is the given type marked as [intrinsic]?
+    // This defaults to false.
+    virtual bool isIntrinsicType(
```

Comment:
Do you also need a method to get the type name, similar to getMethodNameFromMetadata ?

Reply:
At this time, for HW intrinsic work, `isIntrinsicType` is enough to recognize the new SIMD types. That we can distinguish `Vector128` and `Vector256` by type size.
However, if we have to extend the target to `class` (as @AndyAyersMS said) for specially treat more types. I think we need `getTypeNameFromMetadata`.